Replace compilation-error-regexp alists rather than appending to them

ineu opened the pull request:
Current `compilation-errors-regexp-alist*` is great, however it is appended to the original list, making matches too wide. After catching rspec errors, it goes further and runs other matchers which find incorrect lines. For example,
```ruby
/home/ineu/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/htmlcompressor-0.1.2/lib/htmlcompressor/compressor.rb:112: warning: duplicated key at line 132 ignored: :javascript_compressor
/home/ineu/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/htmlcompressor-0.1.2/lib/htmlcompressor/compressor.rb:113: warning: duplicated key at line 133 ignored: :css_compressor
```
It marks these lines as errors as well, though they are really deprecations.
This PR completely replaces predefined alist.

ineu commented:
It also changes regexp to match 'rspec' at the beginning of line. This is also useful to avoid false matches in deprecation warnings like
```ruby
/home/ineu/.rbenv/versions/2.2.0/bin/ruby -I/home/ineu/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/rspec-core-3.1.7/lib:/home/ineu/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/rspec-support-3.1.2/lib /home/ineu/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/rspec-core-3.1.7/exe/rspec /home/ineu/work/foospec/controllers/exit_pages_controller_spec.rb:353

```

dgutov commented:
I like the idea about ditching the default list, but we should take care to still match everything useful.

This fails to highlight the backtraces in the "Failures:" block. Here's how it looks for me:

```
Robe::Sash
  #class_locations
    shows no location for class without methods (FAILED - 1)

Failures:

  1) Robe::Sash#class_locations shows no location for class without methods
     Failure/Error: expect(k.class_locations(nil, nil)).to be_empty
     TypeError:
       exception class/object expected
     # ./lib/robe/sash.rb:28:in `raise'
     # ./lib/robe/sash.rb:28:in `class_locations'
     # ./spec/robe/sash_spec.rb:23:in `block (3 levels) in <top (required)>'

Finished in 0.00148 seconds
1 example, 1 failure

Failed examples:

rspec ./spec/robe/sash_spec.rb:21 # Robe::Sash#class_locations shows no location for class without methods
```

With your patch, `./spec/robe/sash_spec.rb` in the "Failed examples:" section is highlighted (and thus is clickable), but the backtrace above it, isn't.

ineu commented:
@dgutov  Thank you, I'll check it out.

ineu commented:
@dgutov 
Updated, now it matches both backtrace and rspec summary lines.

ineu commented:
Fixed regexp, added tests for it. Couldn't find proper examples on elisp testing, so made it ruby-like :)
Running with
```elisp
(defun my--emacs-lisp-mode-keys ()
  (define-key (current-local-map) (kbd "C-c , v") (lambda () (interactive) (eval-buffer) (ert t)))
	)
(add-hook 'emacs-lisp-mode-hook 'my--emacs-lisp-mode-keys)
```

dgutov commented:
Thanks, look good, but:

1. Please amend the commit not to put closing parens on separate lines.

2. If you replace `(load-file "../rspec-mode.el")` with `(require 'rspec-mode)`, does it fail?

dgutov commented:
By the way, I only have a binding for `eval-buffer`. Then, as I edit tests, I reload their definitions using this command (or just `C-M-x`), and to run tests, just type `M-x ert`.

ineu commented:
> Please amend the commit not to put closing parens on separate lines.
Ok. Is there any styleguide for elisp? I believe I've seen one on gnu.org,
but it is down atm.

> If you replace (load-file "../rspec-mode.el") with (require 'rspec-mode),
does it fail?
No, but it is not suitable for TDD, as "require" would only load feature
once, so to pick changes in the code you'll need to evaluate it manually.

dgutov commented:
> Ok. Is there any styleguide for elisp?

This one is pretty good: https://github.com/bbatsov/emacs-lisp-style-guide

> No, but it is not suitable for TDD, as "require" would only load feature once, so to pick changes in the code you'll need to evaluate it manually.

You might want to get used to `eval-defun` (`C-M-x`) and `M-x eval-buffer`. Running tests from the command line from time to time or also a good practice, but it requires separate configuration.

ineu commented:
Amended. `C-M-x` is fine, though I had to make a little change for it to work: replaced `(setq rspec--test-re)` with `(defun rspec--test-re)`, so this value is not cached in tests.

dgutov commented:
Thanks!
